R/vctr.R

```diff
@@ -387,12 +387,18 @@ xtfrm.vctrs_vctr <- function(x) {
   proxy <- vec_proxy_compare(x)
 
   # order(order(x)) ~= rank(x)
-  if (is.data.frame(proxy)) {
-    order(do.call(base::order, proxy))
-  } else if (is_integer(proxy) || is_double(proxy)) {
+  if (is_integer(proxy) || is_double(proxy)) {
     proxy
-  } else if (is_character(proxy) || is_logical(proxy)) {
-    order(order(proxy))
+  } else {
+    order(order_proxy(proxy))
+  }
+}
+
+order_proxy <- function(proxy) {
```

Comment:
I don't understand why you extracted this out.

Reply:
So that I can do `order_proxy(vec_proxy_compare(df$key))` . Do you see a better way?